.travis.yml

```diff
@@ -1,40 +1,56 @@
+---
+
 language: python
-python:
-- 3.6
-cache: pip
-jobs:
+dist: xenial  # required for Python >= 3.7
+sudo: required
+cache:
+  directories:
+    - "$HOME/.cache/pip"
+    - "$HOME/.pyenv"
+matrix:
   include:
-  - language: python
-    python:
-    - '3.6'
-    cache: pip
-    sudo: required
-    script:
-    - pip install -e . -r requirements.txt -r requirements-tests.txt
-    - grep -Rl packagecloud /etc/apt/ |sudo xargs rm -vf
-    - sudo apt-get update && sudo apt-get install -y nodejs npm && npm install -g dockerfilelint
-    - dockerfilelint ./Dockerfile
-    - flake8 vmware_exporter tests
-    - pytest --cov=. tests/unit
-    after_success:
-    - codecov
-  - language: python
-    python: '3.6'
-    cache: pip
-    sudo: required
-    services:
-    - docker
-    script:
-    - pip install -e . -r requirements.txt -r requirements-tests.txt
-    - pytest tests/integration
-    after_success:
-    - git config --global user.name "semantic-release (via TravisCI)"
-    - git config --global user.email "semantic-release@travis"
-    - pip install python-semantic-release
-    - semantic-release publish
-    - semantic-release changelog --post
+    - name: "Python 3.6 Unit-test"
+      python: 3.6
+    - name: "Python 3.7 Unit-test"
+      python: 3.7
+    - name: "Python 3.7 Docker Unit and Integration-test"
+      python: 3.7
+      services:
+        - docker
+      env:
+        - BUILD_SDIST=true
+        - INTEGRATION_TEST=true
+
+install:
+  - pip install -e . -r requirements.txt -r requirements-tests.txt
+  - grep -Rl packagecloud /etc/apt/ |sudo xargs rm -vf
+  - sudo apt-get update && sudo apt-get install -y nodejs npm
+  - npm install -g dockerfilelint
+  - if [ $INTEGRATION_TEST == "true" ]; then
+      pip install python-semantic-release;
+    fi
+
+script:
+  - dockerfilelint ./Dockerfile
+  - flake8 vmware_exporter tests
+  - pytest --cov=. -v tests/unit
+  - if [ $INTEGRATION_TEST == "true" ]; then
+      pytest tests/integration;
+    else
+      echo "Skipped Integration-Tests";
+    fi
+
+after_success:
+  - codecov
+  - if [ $INTEGRATION_TEST == "true" ]; then
```

Comment:
Feels a bit weird to hang the release pipeline off `INTEGRATION_TEST` - should there be a different variable for the release pipeline?

Or should we have stages and have a release stage that only runs after the test stage?

Reply:
That is basically the same logic that has been put in the origin .travis.yml file. Only other step that was just called "python" was docker + integration tests. I just made it clearer by naming it this way.

Are you available in IRC? Maybe a short chat can clarify why I put things this way

Reply:
There is a vmware_exporter gitter.im channel that all the reviewers are on.

I am happy with the direction this is going in it was just an idea for making it even better. I can do a follow up PR if you aren't sure about stages.